packages/build-info/src/build-systems/package-managers.ts

```diff
@@ -0,0 +1,87 @@
+import { isNpmBuildScript, isNpmDevScript } from '../get-commands.js'
+import { PkgManager } from '../package-managers/detect-package-manager.js'
+
+import { BaseBuildTool, Command } from './build-system.js'
+
+export class PNPM extends BaseBuildTool {
+  id = 'pnpm'
+  name = 'PNPM'
+  runFromRoot = true
+
+  // will be called after the framework detection has run
+  async getCommands(packagePath: string): Promise<Command[]> {
+    const { name, scripts } = await this.project.fs.readJSON(
+      this.project.resolveFromPackage(packagePath, 'package.json'),
+    )
+    if (scripts && Object.keys(scripts).length > 0) {
+      return Object.entries(scripts).map(([scriptName, value]) => ({
```

Comment:
nit: I find these lines a bit hard to parse, where you have a `map` with an inline function that returns an object with a property that returns a ternary operator with two function calls. 😅